sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Infrastructure/ProcessorTest.cs

```diff
@@ -3,61 +3,169 @@
-            };
+            // Create the Event Hub and seed it with events.
 
-            if (options.CacheEventCount.HasValue)
+            Scope = await EventHubScope.CreateAsync(Options.PartitionCount).ConfigureAwait(false);
+
+            await using var producer = new EventHubProducerClient(TestEnvironment.EventHubsConnectionString, Scope.EventHubName);
+            await SeedEventsToBeReadAsync(producer, Options.EventBodySize, Options.EventSeedCount).ConfigureAwait(false);
+
+            // Ensure that the Blob storage container for processor data exists.
+
+            await _checkpointStore.CreateIfNotExistsAsync().ConfigureAwait(false);
+
+            // Create the processor client.
+
+            var clientOptions = new EventProcessorClientOptions { LoadBalancingStrategy = Options.LoadBalancingStrategy };
+
+            if (Options.CacheEventCount.HasValue)
             {
-                clientOptions.CacheEventCount = options.CacheEventCount.Value;
+                clientOptions.CacheEventCount = Options.CacheEventCount.Value;
             }
 
-            if (options.MaximumWaitTimeMs.HasValue)
+            if (Options.MaximumWaitTimeMs.HasValue)
             {
-                clientOptions.MaximumWaitTime = TimeSpan.FromMilliseconds(options.MaximumWaitTimeMs.Value);
+                clientOptions.MaximumWaitTime = TimeSpan.FromMilliseconds(Options.MaximumWaitTimeMs.Value);
             }
 
-            if (options.PrefetchCount.HasValue)
+            if (Options.PrefetchCount.HasValue)
             {
-                clientOptions.PrefetchCount = options.PrefetchCount.Value;
+                clientOptions.PrefetchCount = Options.PrefetchCount.Value;
             }
 
             EventProcessorClient = new EventProcessorClient(
                 _checkpointStore,
-                EventHubConsumerClient.DefaultConsumerGroupName,
-                EventHubsTestEnvironment.Instance.EventHubsConnectionString,
-                EventHubsTestEnvironment.Instance.EventHubNameOverride,
+                Scope.ConsumerGroups[0],
+                TestEnvironment.EventHubsConnectionString,
+                Scope.EventHubName,
                 clientOptions);
         }
 
-        public override async Task SetupAsync()
+        /// <summary>
+        ///   Performs the tasks needed to initialize and set up the environment for the test scenario.
+        ///   This setup will take place once for each instance, running before the global cleanup is
+        ///   run.
+        /// </summary>
+        ///
+        public override async Task CleanupAsync()
         {
-            await base.SetupAsync();
-            await _checkpointStore.CreateIfNotExistsAsync();
+            try
+            {
+                await _checkpointStore.DeleteIfExistsAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                await Scope.DisposeAsync().ConfigureAwait(false);
```

Comment:
NIT: Should `Scope.DisposeAsync()` be called from `CleanupAsync()` or `DisposeAsyncCore()`?

https://github.com/Azure/azure-sdk-for-net/blob/c56d65f9357b7c5f64aa7ed2fa31f0098480ee74/common/Perf/Azure.Test.Perf/PerfTestBase.cs#L87

Reply:
The existing tests in the core library are doing the scope manipulation in GlobalSetup/Setup and the teardown in the GlobalCleanup/Cleanup.  I went with the same pattern here for consistency.

I don't have strong opinions, but I'd prefer to keep it in Cleanup since we don't have the option of a GlobalDispose.